addons/sale_project/models/sale_order_line.py

```diff
@@ -416,3 +416,9 @@ def _get_action_per_item(self):
             :returns: Dict containing id of SOL as key and the action as value
         """
         return {}
+
+    def _prepare_procurement_values(self, group_id=False):
+        values = super()._prepare_procurement_values(group_id=group_id)
+        if self.project_id:
+            values['project_id'] = self.order_id.project_id.id
```

Comment:
sale.order.line also have `project_id` which seems more precise than the SO one

Reply:
@Whenrow Actually, the `project_id` field in SOL is more used to store the project generated by the SOL, so, I think it is better to use the one defined in SO as we did here. 🙂

Reply:
@xavierbol ok I guess this `project_id` is only relevant with service product on SO line

Reply:
@Whenrow yep exactly. 🙂